### PR TITLE
Trim common whitespace after evaluating command block

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: src/main/scala/wdlTools/util/Util.scala [hunk 1]
```diff
@@ -193,36 +195,44 @@ object Util {
...
-    val nonWsRegex = "^(\\s.*)(.*)$".r
-    val (lineOffset, content) = lines.foldLeft((0, Vector.empty[(Int, String)])) {
+  def stripLeadingWhitespace(s: String,
+                             ignoreEmptyLast: Boolean = false,
+                             lineSep: String = System.lineSeparator()): (Int, Int, String) = {
+    val lines = s.split("\r\n?|\n")
+    val wsRegex = "^([ \t]*)$".r
+    val nonWsRegex = "^([ \t]*)(.+)$".r
+    val (lineOffset, content) = lines.foldLeft((0, Vector.empty[(String, String)])) {
       case (l, wsRegex(_)) if l._2.isEmpty => (l._1 + 1, l._2)
```
Comment (orodeh) [inline src/main/scala/wdlTools/util/Util.scala:209]: Could you please replace the `_1, _2` shorthand with meaningful variable names? It is harder to understand this way.

File: src/main/scala/wdlTools/util/Util.scala [hunk 2]
```diff
@@ -193,36 +195,44 @@ object Util {
...
-      case other                           => throw new Exception(s"Unexpected non-matching line ${other}")
+  def stripLeadingWhitespace(s: String,
+                             ignoreEmptyLast: Boolean = false,
+                             lineSep: String = System.lineSeparator()): (Int, Int, String) = {
+    val lines = s.split("\r\n?|\n")
+    val wsRegex = "^([ \t]*)$".r
+    val nonWsRegex = "^([ \t]*)(.+)$".r
+    val (lineOffset, content) = lines.foldLeft((0, Vector.empty[(String, String)])) {
+      case ((lineOffset, content), wsRegex(_)) if content.isEmpty => (lineOffset + 1, content)
+      case ((lineOffset, content), nonWsRegex(ws, txt))           => (lineOffset, content :+ (ws, txt))
```
Comment (orodeh) [inline src/main/scala/wdlTools/util/Util.scala:210]: I didn't know you could this in scala `nonWsRegex(ws, txt)`. Very nice.

File: src/main/scala/wdlTools/util/Util.scala [hunk 3]
```diff
@@ -193,36 +195,44 @@ object Util {
...
-             strippedLines.zip(wsLengths).map {
-               case (line, wsLength) if wsLength > colOffset =>
-                 (" " * (wsLength - colOffset)) + line
-               case (line, _) => line
-             }
-           }
-       ).mkString(System.lineSeparator()))
+      val (whitespace, strippedLines) = (
+          if (ignoreEmptyLast && content.nonEmpty && content.last._2.trim.isEmpty) {
+            content.slice(0, content.length - 1)
```
Comment (orodeh) [inline src/main/scala/wdlTools/util/Util.scala:218]: I think you can just do `content.tail` here

File: src/test/scala/wdlTools/eval/EvalTest.scala [hunk 4]
```diff
@@ -259,34 +260,47 @@ class EvalTest extends AnyFlatSpec with Matchers with Inside {
...
-    command.trim shouldBe "hello"
+    command shouldBe "hello"
+  }
+
+  it should "strip common indent" taggedAs Edge in {
+    val command = evalCommand("indented_command.wdl")
+    command shouldBe " echo 'hello Steve'\necho 'how are you, Steve?'\n   echo 'goodbye Steve'"
+  }
+
+  it should "strip common indent in python heredoc" taggedAs Edge in {
```
Comment (orodeh) [inline src/test/scala/wdlTools/eval/EvalTest.scala:296]: Great, this is the kind of use I am worried about. Now we have a unit test for it.

File: src/main/scala/wdlTools/util/Util.scala (hunk 3 above)
Comment (jdidion) [reply src/main/scala/wdlTools/util/Util.scala:218]: tail is everything after the first item. I'm trying to get everything except the last item.

**Comment by orodeh** [comment]
Sorry, you are right. In that case, it is `dropRight(1)`

**Pull request merged**